src/Microsoft.AspNet.Server.Kestrel/Infrastructure/TaskUtilities.cs

```diff
@@ -12,5 +15,72 @@ public static class TaskUtilities
 #else
         public static Task CompletedTask = Task.FromResult<object>(null);
 #endif
+
+        private static WaitCallback _runAction = (o) =>
+        {
+            try
+            {
+                ((Action)o)();
+            }
+            catch (Exception)
+            {
+                // log with a static logger in some way?
+            }
+        };
+
+        private static WaitCallback _completeTcs = (o) =>
+        {
+            try
+            {
+                ((TaskCompletionSource<object>)o).SetResult(null);
```

Comment:
Given that we aren't likely to see the callback added very soon, I think we might want something like this.

The main thing preventing me from merging this is the fact that it doesn't log the exceptions.